sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -4228,13 +4228,15 @@ pi_result piKernelGetGroupInfo(pi_kernel Kernel, pi_device Device,
   std::shared_lock<pi_shared_mutex> Guard(Kernel->Mutex);
   switch (ParamName) {
   case PI_KERNEL_GROUP_INFO_GLOBAL_WORK_SIZE: {
-    // TODO: To revisit after level_zero/issues/262 is resolved
```

Comment:
Do you have a link to this issue?

Reply:
Glad you've asked. I've tried to find it in `[level_zero](https://github.com/oneapi-src/level-zero/issues)` and [L)-spec](https://github.com/oneapi-src/level-zero-spec/issues). Couldn't find an issue related to this. There were any issue with tags greater than 150s. 

Looked into compute-RT and found this: https://github.com/intel/compute-runtime/issues/262. Didn't really look related.

Reply:
The issue was closed before the remaining open ones were moved to a new location. So we will not be able to find what it was, I am afraid. The change looks good to me but I'd want @jandres742 to confirm.

Reply:
why are we multiplying? what is the meaning PI_KERNEL_GROUP_INFO_GLOBAL_WORK_SIZE?

In L0 we have maxTotalGroupSize, would that be what we need here?
https://spec.oneapi.io/level-zero/latest/core/api.html?highlight=maxtotalgroupsize#_CPPv4N30ze_device_compute_properties_t17maxTotalGroupSizeE

Reply:
@jandres742 Thanks for pointing to the documentation. These are the wordings from the spec `The exact semantics of this descriptor are defined by each SYCL backend specification, but the intent is to return the kernel’s maximum global work size.` 

Since the return is per-dimension (i.e., a `sycl::range<3>`), I was multiplying for each dimension. I looked into `maxTotalGroupSize` and that looked like for "max size per compute work-group" and it might not be a functionally equivalent one. The functionality was for the "global maximum iteration limits per dimension"

Reply:
thanks @abagusetty . I read the definition of CL_KERNEL_GLOBAL_WORK_SIZE , which would be equivalent, and proposed change seems correct.

FYI: I dont see CL_KERNEL_GLOBAL_WORK_SIZE implemented in Intel OCL GPU driver, so don't know what we are returning for the OpenCL GPU backend.